lib/rules/color-hex-case/index.js

```diff
@@ -26,68 +30,56 @@ function rule(expectation, options, context) {
 		}
 
 		root.walkDecls((decl) => {
-			const declString = blurFunctionArguments(decl.toString(), 'url');
-			const fixPositions = [];
+			const parsedValue = valueParser(getValue(decl));
+			let needsFix = false;
 
-			styleSearch({ source: declString, target: '#' }, (match) => {
-				const hexMatch = /^#[0-9A-Za-z]+/.exec(declString.substr(match.startIndex));
+			parsedValue.walk((node) => {
+				const { type, value } = node;
 
-				if (!hexMatch) {
-					return;
+				// Only traverse a function if it's a color function.
```

Comment:
Let's pull this out into a self-documenting function, e.g.:

```js
if (isIgnoredFunction(node)) return false;

// at the bottom of file
function isIgnoredFunction({type, value}) {
  return type === 'function' && IGNORED_FUNCTIONS.has(value.toLowerCase())
}
```

The comment is redundant then.